### ISSUE TITLE
1 - ArgIdx in Sema::AddOverloadCandidate looks like it overflows

### ISSUE THREAD

**Issue by shafik** [llvm/llvm-project#135086]
Static analysis flagged the `1 - ArgIdx` as possibly leading to overflow:

https://github.com/llvm/llvm-project/blob/f75dce43ea0a98c0ade916cf3ffcc7548f569821/clang/lib/Sema/SemaOverload.cpp#L7196

Looking at somewhat related code maybe it was meant to be `Args.size() - 1 - ArgIdx`.

**Comment by shafik** [comment]
CC @zygoloid since this was introduced via 974c8b7e2fde550fd87850d50695341101c38c2d

**Comment by zygoloid** [comment]
`OverloadCandidateParamOrder::Reversed` is only used for overloaded comparison operators, for which the number of arguments is always 2 here. Maybe we could add something like
```c++
assert(PO != OverloadCandidateParamOrder::Reversed || Args.size() == 2);
```
